### PR TITLE
[NEW] Configurable Volume for Notifications #6087

### LINKED ISSUE #6087: Configurable Volume for Notifications
Using html5's `audioelem.volume`

### PR DESCRIPTION
**Pull request by lindoelio**
@RocketChat/core 

Closes #6087

image

### CODE REVIEW

File: packages/rocketchat-lib/server/models/Users.js [hunk 1]
```diff
@@ -397,7 +397,6 @@ class ModelUsers extends RocketChat.models._Base {
 				requirePasswordChangeReason
 			}
 		};
-
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/models/Users.js]: Please, do not introduce unnecessary changes

File: packages/rocketchat-lib/server/startup/settings.js [hunk 2]
```diff
@@ -70,7 +70,6 @@ RocketChat.settings.addGroup('Accounts', function() {
 		type: 'boolean',
 		'public': true
 	});
-
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/startup/settings.js]: Please, do not introduce unnecessary changes

File: packages/rocketchat-theme/client/vendor/jscolor.js [hunk 3]
```diff
@@ -619,7 +619,7 @@ var jsc = {
 
 		switch (controlName) {
 		case 'pad':
-			// if the slider is at the bottom, move it up
```
Comment (rodrigok) [inline packages/rocketchat-theme/client/vendor/jscolor.js]: Why did you change these comments?

File: packages/rocketchat-i18n/i18n/es.i18n.json [hunk 4]
```diff
@@ -872,6 +872,7 @@
   "Nothing_found": "No se encontró nada",
   "Notification_Duration": "Duración de la notificación",
   "Notifications": "Notificaciones",
+  "Notifications_Sound_Volume": "Notifications sound volume",
```
Comment (rodrigok) [inline packages/rocketchat-i18n/i18n/es.i18n.json]: This is not correct

File: .meteor/packages [hunk 5]
```diff
@@ -173,3 +174,4 @@ yasinuslu:blaze-meta
 deepwell:bootstrap-datepicker2
 rocketchat:postcss
 dynamic-import@0.1.1
+fourseven:scss
```
Comment (karlprieb) [inline .meteor/packages]: We use only cssnext as CSS post-processor.

File: packages/rocketchat-i18n/i18n/sq.i18n.json [hunk 6]
```diff
@@ -748,6 +748,7 @@
   "Nothing": "asgjë",
   "Nothing_found": "Asgjë për të gjetur",
   "Notifications": "Njoftime",
+  "Notifications_Sound_Volume": "Notifications sound volume",
```
Comment (karlprieb) [inline packages/rocketchat-i18n/i18n/sq.i18n.json]: This is not correct

File: packages/rocketchat-livechat/app/client/lib/_visitor.js [hunk 7]
```diff
@@ -58,7 +58,10 @@ this.visitor = new class {
 
 				// notification sound
 				if (Session.equals('sound', true) && msg.u._id !== Meteor.userId()) {
-					$('#chatAudioNotification')[0].play();
+					const audioVolume = Meteor.user().settings.preferences.notificationsSoundVolume || 100;
+					const audio = $('#chatAudioNotification')[0];
```
Comment (karlprieb) [inline packages/rocketchat-livechat/app/client/lib/_visitor.js]: you can use `document.getElementById` here, it's faster if you won't use jQuery.

File: packages/rocketchat-slider/rocketchat-slider.scss [hunk 8]
```diff
@@ -0,0 +1,112 @@
+$shade-10: #04436a !default;
```
Comment (karlprieb) [inline packages/rocketchat-slider/rocketchat-slider.scss]: We don't use scss, just cssnext and less(we will remove all less styles in the future). You need to add your style on rocketchat-theme package and set your custom colors there on variables.js and colors.less files.

File: .meteor/packages (hunk 5 above)
Comment (lindoelio) [inline .meteor/packages]: Ok. SCSS removed.

File: packages/rocketchat-i18n/i18n/es.i18n.json (hunk 4 above)
Comment (lindoelio) [inline packages/rocketchat-i18n/i18n/es.i18n.json]: Fixed!

File: packages/rocketchat-i18n/i18n/sq.i18n.json (hunk 6 above)
Comment (lindoelio) [inline packages/rocketchat-i18n/i18n/sq.i18n.json]: Fixed!

File: packages/rocketchat-lib/server/models/Users.js (hunk 1 above)
Comment (lindoelio) [inline packages/rocketchat-lib/server/models/Users.js]: Sorry. Fixed!

File: packages/rocketchat-lib/server/startup/settings.js (hunk 2 above)
Comment (lindoelio) [inline packages/rocketchat-lib/server/startup/settings.js]: Sorry. Fixed!

File: packages/rocketchat-livechat/app/client/lib/_visitor.js (hunk 7 above)
Comment (lindoelio) [inline packages/rocketchat-livechat/app/client/lib/_visitor.js]: Thanks for your feedback. Fixed!

File: packages/rocketchat-slider/rocketchat-slider.scss (hunk 8 above)
Comment (lindoelio) [inline packages/rocketchat-slider/rocketchat-slider.scss]: Thanks for your feedback and explanation. I've done as you suggested. Fixed!

File: packages/rocketchat-theme/client/vendor/jscolor.js (hunk 3 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/vendor/jscolor.js]: Sorry. I've struggled by my find-replace :-) Fixed!

File: packages/rocketchat-theme/server/colors.less [hunk 9]
```diff
@@ -911,3 +911,50 @@ label.required::after {
 		opacity: inherit;
 	}
 }
+
+/** ----------------------------------------------------------------------------
+ * Input Range Slider
+ */
+
+.range-slider__range {
```
Comment (karlprieb) [inline packages/rocketchat-theme/server/colors.less]: We already have classes for these colors, can you add those classes on your template?

File: packages/rocketchat-ui-account/client/accountPreferences.js [hunk 10]
```diff
@@ -1,5 +1,6 @@
 /*globals defaultUserLanguage, KonchatNotification */
 import toastr from 'toastr';
+
```
Comment (karlprieb) [inline packages/rocketchat-ui-account/client/accountPreferences.js]: can you remove that unnecessary change?

File: packages/rocketchat-theme/client/imports/slider.css [hunk 11]
```diff
@@ -0,0 +1,72 @@
+*,
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: we already have this selector on `base.css` file

File: packages/rocketchat-theme/client/imports/slider.css [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+	left: -7px;
+	width: 0;
+	height: 0;
+	border-top: 7px solid;
+	border-right: 7px solid;
+	border-bottom: 7px solid;
+	content: '';
+}
+
+::-moz-range-track {
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: can you make a selector that work only on the slider?

File: packages/rocketchat-theme/client/imports/slider.css [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+	border: 0;
+}
+
+input::-moz-focus-inner,
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: can you make a selector that work only on the slider?

File: .meteor/versions [hunk 14]
```diff
@@ -44,6 +44,7 @@ email@1.2.3
 emojione:emojione@3.0.3
 facebook-oauth@1.3.2
 fastclick@1.0.13
+fourseven:scss@4.5.4
```
Comment (karlprieb) [inline .meteor/versions]: can you remove that package?

File: packages/rocketchat-theme/client/imports/slider.css [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+	box-sizing: border-box;
+}
+
+.range-slider {
+	margin: 0 0 0 0%;
+	width: 100%;
+}
+
+.range-slider__range {
+	-webkit-appearance: none;
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: we already use autoprefixer on css build, so you can remove these browser prefixes.

File: packages/rocketchat-theme/client/imports/slider.css [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+	width: calc(100% - (73px));
+	height: 10px;
+	border-radius: 5px;
+	outline: none;
+	padding: 0;
+	margin: 0;
+}
+
+.range-slider__range::-webkit-slider-thumb {
+	-webkit-appearance: none;
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: we already use autoprefixer on css build, so you can remove these browser prefixes.

File: packages/rocketchat-theme/client/imports/slider.css [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+	width: 20px;
+	height: 20px;
+	border-radius: 50%;
+	cursor: pointer;
+	-webkit-transition: background 0.15s ease-in-out;
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: we already use autoprefixer on css build, so you can remove these browser prefixes.

File: packages/rocketchat-theme/client/imports/slider.css [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.range-slider {
+	margin: 0 0 0 0%;
+	width: 100%;
+}
+
+.range-slider__range {
+	-webkit-appearance: none;
+	width: calc(100% - (73px));
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: you can use `calc(100% - 73px)`

File: packages/rocketchat-theme/client/imports/slider.css [hunk 19]
```diff
@@ -0,0 +1,72 @@
+*,
+*::before,
+*::after {
+	box-sizing: border-box;
+}
+
+.range-slider {
+	margin: 0 0 0 0%;
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: you can use `margin: 0`

File: packages/rocketchat-theme/client/imports/slider.css [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+	transition: background 0.15s ease-in-out;
+}
+
+.range-slider__range::-moz-range-thumb {
+	width: 20px;
+	height: 20px;
+	border: 0;
+	border-radius: 50%;
+	cursor: pointer;
+	-webkit-transition: background 0.15s ease-in-out;
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/slider.css]: we already use autoprefixer on css build, so you can remove these browser prefixes.

File: packages/rocketchat-theme/server/colors.less (hunk 9 above)
Comment (lindoelio) [inline packages/rocketchat-theme/server/colors.less]: Fixed!

File: packages/rocketchat-ui-account/client/accountPreferences.js (hunk 10 above)
Comment (lindoelio) [inline packages/rocketchat-ui-account/client/accountPreferences.js]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 11 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 19 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 15 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 18 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 16 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 17 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 20 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 12 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 13 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: .meteor/versions (hunk 14 above)
Comment (lindoelio) [inline .meteor/versions]: Fixed!

File: packages/rocketchat-push-notifications/client/views/pushNotificationsFlexTab.js [hunk 21]
```diff
@@ -255,29 +255,37 @@ Template.pushNotificationsFlexTab.events({
 
 	'click [data-play]'(e) {
 		e.preventDefault();
+
 		let audio = $(e.currentTarget).data('play');
+
+		if (!audio || audio === 'none') {
+			audio = Meteor.user().settings.preferences.newMessageNotification || 'chime';
```
Comment (lindoelio) [inline packages/rocketchat-push-notifications/client/views/pushNotificationsFlexTab.js]: Fixed!

File: packages/rocketchat-ui-account/client/accountPreferences.js (hunk 10 above)
Comment (lindoelio) [inline packages/rocketchat-ui-account/client/accountPreferences.js]: Fixed!

File: packages/rocketchat-theme/client/imports/slider.css (hunk 11 above)
Comment (lindoelio) [inline packages/rocketchat-theme/client/imports/slider.css]: Fixed!

File: packages/rocketchat-ui/client/lib/notification.js [hunk 22]
```diff
@@ -79,15 +79,27 @@ const KonchatNotification = {
...
-			if (sub && sub.audioNotification !== 'none') {
-				if (sub && sub.audioNotification) {
-					const [audio] = $(`audio#${ sub.audioNotification }`);
-					return audio && audio.play && audio.play();
-				} else if (newMessageNotification !== 'none') {
-					const [audio] = $(`audio#${ newMessageNotification }`);
-					return audio && audio.play && audio.play();
+
+			if (user && user.settings && user.settings.preferences) {
+				const newMessageNotification = user.settings.preferences.newMessageNotification || 'chime';
```
Comment (rodrigok) [inline packages/rocketchat-ui/client/lib/notification.js]: If there is no `user.settings.preferences` then there will not have notifications?

File: packages/rocketchat-livechat/app/client/lib/_visitor.js [hunk 23]
```diff
@@ -58,7 +58,10 @@ this.visitor = new class {
 
 				// notification sound
 				if (Session.equals('sound', true) && msg.u._id !== Meteor.userId()) {
-					$('#chatAudioNotification')[0].play();
+					const audioVolume = Meteor.user().settings.preferences.notificationsSoundVolume || 100;
```
Comment (rodrigok) [inline packages/rocketchat-livechat/app/client/lib/_visitor.js]: Can you check the properties here too?

Comment (lindoelio) [inline packages/rocketchat-livechat/app/client/lib/_visitor.js]: Checked!

File: packages/rocketchat-ui/client/lib/notification.js (hunk 22 above)
Comment (lindoelio) [inline packages/rocketchat-ui/client/lib/notification.js]: Fixed!
